src/NzbDrone.Core/Notifications/Join/JoinProxy.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using FluentValidation.Results;
+using NLog;
+using RestSharp;
+using NzbDrone.Core.Rest;
+using NzbDrone.Common.Serializer;
+
+namespace NzbDrone.Core.Notifications.Join
+{
+    public interface IJoinProxy
+    {
+        void SendNotification(string title, string message, JoinSettings settings);
+        ValidationFailure Test(JoinSettings settings);
+    }
+
+    public class JoinProxy : IJoinProxy
+    {
+        private readonly Logger _logger;
+        private const string URL = "https://joinjoaomgcd.appspot.com/_ah/api/messaging/v1/sendPush?";
+
+        public JoinProxy(Logger logger)
+        {
+            _logger = logger;
+        }
+
+        public void SendNotification(string title, string message, JoinSettings settings)
+        {
+            var request = new RestRequest(Method.GET);
+
+            try
+            {
+                SendNotification(title, message, request, settings);
+            }
+            catch (JoinException ex)
+            {
+                _logger.Error(ex, "Unable to send Join message.");
+                throw new JoinException("Unable to send Join notifications. " + ex.Message);
+            }
+        }
+
+        public ValidationFailure Test(JoinSettings settings)
+        {
+
+            const string title = "Test Notification";
+            const string body = "This is a test message from Sonarr.";
+            try
+            {
+                SendNotification(title, body, settings);
+                return null;
+            }
+            catch (JoinException ex)
```

Comment:
Since we're throwing JoinExceptions for any exception this could be misleading if the test fails for another reason (Join servers are down for example). We'll still want to catch a JoinException, but we should also have a JoinAuthentication Exception that inherits from JoinException that we can throw when we're sure its an Authentication issue.